### DISCUSSION TITLE
nautilus: cannot create new thumbnails (bwrap/userns)

### DISCUSSION THREAD

**Discussion by Froggy232** [netblue30/firejail#4135] [Q&A]
Hi,

Sorry to bother you again, this time it's for a very little problem : since I applied firecfg, nautilus is not able to create thumbnails anymore. I still can see the previous ones, but all new files just stay with the default icons.

Would someone have an idea on how to fix the issue? I haven't find anything on internet.

It's not a big problem, but I would love to get back this feature.
Thanks you

**Comment by rusty-snake** [comment]
What's your firejail version? Or precisely asked was ffmpeg got firejailed?

**Reply by Froggy232** [reply]
Hi,
Thanks for your answer!

I use the `0.9.64.4-1` version, I use Archlinux so it's the latest one.
There is a ffmpeg and ffmpegthumbnailer in `/usr/local/bin`, but if I remove them it don't change anything. Also, the problem is here for all thumbnails, including the ones which are not dependant on ffmpeg, like PDF.

Thanks for your help!

**Reply by rusty-snake** [reply]
What programs are firejail when you nautilus tries to generate thumbnails? (`firejail --list` or `sudo firemon`).

**Reply by rusty-snake** [reply]
FWIW: Newer versions of nautilus use bwrap to sandbox thumbnails. This means (a) you can not use firejail for the thumbnailers and (b) you can not use firejail for nautilus.

**Reply by Froggy232** [reply]
There is absolutely no new entry on `sudo firemon` when I go to a folder with no-thumbnailed files, it stay the same.
All the software writen in firemon have no relation to thumbnail I think (firefox and evolution), except maybe `xdg-dbus-proxy`?
I can post the total output of `sudo firemon` if you want.
Thanks for your help!

**Reply by Froggy232** [reply]
> FWIW: Newer versions of nautilus use bwrap to sandbox thumbnails. This means (a) you can not use firejail for the thumbnailers and (b) you can not use firejail for nautilus.

Thanks, so I need to remove `ffmpegthumbnailer` from `/usr/local/bin` and blacklist it from `firecfg.conf`?

**Reply by rusty-snake** [reply]
yes

**Reply by Froggy232** [reply]
Thanks, I will remove it. Do you know if there is other firejail rule I need to remove?
I suspect a problem from apparmor, I will keep the thread updated.

Thanks you!

**Reply by Froggy232** [reply]
So, the root cause wasn't firejail but linux-hardened : as it disable user-namespace, one need to replace bubblewrap with bubblewrap-suid (https://bugs.archlinux.org/task/63762).

Still, the PDF thumbnail doesn't work if I keep evince-thumbnailer firejailed.
I have simply commented all thumbnailers from firecfg.config for the moment, but if someone know what to add to .local files, I would be happy to try.

Best regards

**Reply by rusty-snake** [reply]
bubblewrap *always* sets the `NO_NEW_PRIVS` prctl which means firejails suid-bit is ignored and it will fail. There's nothing that can fix this by adding it to a .local. You can not use bubblewrap inside firejail and you can not use firejail inside bubblewrap. Double sandboxing usually makes more pain then it's worth. The thumbnailers are already sandboxed then.

That still leaves us with the question of how to proceed with it. Remove all thumbnailers from firecfg.config and reduce security of other filemanagers (except dolphin) or keep them and break nautilus? Opinions?

**Reply by Froggy232** [reply]
Right, I understand.
Personally, I have disable thumbnailers from `firecfg.config` as I only use nautilus, and as thumbnail are important to my workflow anyway, but I haven't the knowledge to have an opinon on a sane default TBH.
Thanks for your help, as always!
Have a nice day

**Reply by ghost** [reply]
> That still leaves us with the question of how to proceed with it. Remove all thumbnailers from firecfg.config and reduce security of other filemanagers (except dolphin) or keep them and break nautilus? Opinions?

Personally I've always found it very `tedious` to do daily work/pleasure with a firejailed filemanager. Things tend to break left and right all the time and debugging/fixing all these (relatively minor) glitches lead to `overly` complex .local files, loss of time-hair-sanity etcetera. IMO a _reasonably secure_ middleground is NOT to sandbox filemanagers BUT rely on _tightly_ sandboxed individual applications used under the hood anyway - so preferably in whitelisting mode. When it comes to thumbnailing I would rather break nautilus than dropping functional apps from firecfg.config. But I'm old-skool (very) and tend to use filemanagers in detailed list-view mode anyway. If not seeing a big thumbnail is the price to pay for added (sense of) security, I'm happy to do so...

What's more important here IMO is that to inform/advise users on how to drop specific apps from being firecfg'ed by default needs improving - something I know @rusty-snake is well aware of / put extra work in - and probably the raison-d'etre for [firecfg.py](https://github.com/rusty-snake/firecfg.py) in the first place.
